src/useValidation/useValidation.ts

```diff
@@ -0,0 +1,107 @@
+import { ComponentInternalInstance, computed, ComputedRef, getCurrentInstance, inject, onMounted, onUnmounted, ref, Ref, watch } from 'vue'
+import { ValidationAbortedError } from './ValidationAbortedError'
+import { ValidationRuleExecutor } from './ValidationExecutor'
+import { NoInfer } from '@/types/generics'
+import { MaybePromise, MaybeRef } from '@/types/maybe'
+import { isUseValidationObserver, UseValidationObserver, ValidationObserverUnregister, VALIDATION_OBSERVER_INJECTION_KEY } from '@/useValidationObserver/useValidationObserver'
+import { getSymbolForInjectionKey } from '@/utilities/symbols'
+
+const USE_VALIDATION_SYMBOL: unique symbol = Symbol('UseValidationSymbol')
```

Comment:
This symbol is used only for the validation typeguard, yeah? Is there another use I'm missing?

Reply:
Correct. This is a technique I stole from vue's source code. By creating a `unique symbol` and not exporting it you can check for it within this project but it will never show up in the types when you use this composition. Which produced a much simpler and much safer type guard rather than checking for the presence of specific keys.

Reply:
Following a conversation with @stackoverfloweth these guards and symbols ended up being removed. We realized ultimately the guards were not needed.

Reply:
Way ahead of me, thanks for the update 👍🏻